server/server/middlewares/userAuth.js

```diff
@@ -1 +1,54 @@
-// 
\ No newline at end of file
+/***********************************ramah */
+const userModel = require("../models/user");
+const blackListModel = require("../models/blackList");
+const jwt = require("jsonwebtoken");
+const cookie = require('cookie');
+
+// For a user who's logged in
+const ensureAuth = async (req, res, next) => {
+  try {
+    if (req.isAuthenticated()) {
+      return next();
+    }; 
+
+    const authHeader = req.headers["cookie"];
+
+    if(!authHeader){
+      return res.sendStatus(401);
+    };
+    
+    const cookies = cookie.parse(authHeader);
+    const accessToken = cookies["SessionID"];  
+
+    if (!accessToken) {
+     return res.sendStatus(401);
+   };
+   
+   const checkIfBlackListed = await blackListModel.findOne({ token:  accessToken});
```

Comment:
should the name start with a capital letter for model name

Reply:
Done